### ISSUE TITLE
Ractive 0.7 doesn't allow subsequent output?

### ISSUE THREAD

**Issue by matthew-dean** [ractivejs/ractive#1934]
I just upgraded from 0.6 to 0.7. I got this error:
```
"Uncaught ParseError: Expected closing delimiter '}}' after reference at line 22 character 40:
<div class="message-item {{ type }} {{ new }}">
```
I changed nothing with the template. But apparently having two mustaches in a row fails in 0.7.2?

I reset to 0.6.1 and no longer have the error.

**Comment by evs-chris** [comment]
The issue is actually the reference named `new`. The parser got a pretty big refactor for the 0.7 release. The general policy is that javascript keywords are disallowed where they would be invalid in a regular javascript expression (freestanding new in this case).

**Comment by Madgvox** [comment]
Would this be a `better error` situation or does that error make enough sense in context?

**Comment by fskreuz** [comment]
There's nothing wrong with the template (given that example), which makes `Expected closing delimiter '}}'` error rather... confusing. If the main source of the error is `new`, then it should throw at `new`. Probably something like "[keyword] is a reserved keyword"

---

Reminds me of how JS throws "Unexpected identifier" errors when you miss a comma in an object literal. It throws at the next key, and not where the comma was missing. Though correct, but still confusing.

```
var foo = {
  bar: 'baz'      // missing comma here
  hello: 'world'  // JS throws here
};
```

**Comment by YRathod** [comment]
keywords seems to be allowed with dots - {{ .new }} is not throwing an error. 
Related issue - https://github.com/ractivejs/ractive/issues/1497

There should some meaningful error verbiage required.

**Comment by matthew-dean** [comment]
Ah, I see. I thought references were essentially treated as quoted key names. Unless you're saying that new is allowed as part of an expression now? What's the expression conflict?

**Comment by Madgvox** [comment]
`new` by itself is [not allowed](http://docs.ractivejs.org/latest/expressions#valid-expressions) as part of an expression. However, it can't be completely banned, because you can assign it as an object property:

```js
var foo = {};
foo['new'] = 'bar';
// no errors, 'new' is a string
// then you can access it using dot notation, just like in ractive
foo.new;      // 'bar'
```

Since expressions allow basic javascript to be executed in them, they must act like javascript does. Therefore, a lone `new` operator is not considered a reference. You must retrieve it using dot notation. This quirk is not something that ractive does, it's a quirk with the way javascript objects work.

---

@fskreuz I have to agree with you there, but that error doesn't really tell the person that they need to 'escape' the `new` in order for it not to get parsed as an operator. If I saw that I'd probably go and rename the variable `new` instead of prefix it with a dot. Although renaming it is arguably the better solution since having a variable named `new` is probably not the best idea anyway.

**Comment by matthew-dean** [comment]
I went back to 0.6.1 because of 0.7.2 instability. 0.6.2 triggers an error on ```.new``` (with the dot), but does so at the dot position, which yes, is a better error.

**Comment by Rich-Harris** [comment]
@matthew-dean when you say 'instability', are you referring solely to this issue or to other things?

**Comment by matthew-dean** [comment]
@Rich-Harris Sorry about the ambiguity there. What happened for me with 0.7 is that I suddenly had all sorts of rendering errors in my UI. What I later figured out is that I had ambiguous references in my templates, and for whatever reason, 0.6 was more fault-tolerant than 0.7 to recursion. Which may not be a bad thing. Having more consistent errors in 0.7 (when I switched back again after I saw that the errors happened in both places, just less often in 0.6) meant that I figured out it was an ambiguous key reference problem, leading to recursion.

I do have another major problem, that's persisted across both versions, but I'll try to find the time to post that as a new issue.
